### PR TITLE
Custom email address model

### PR DESCRIPTION
**Pull request by jakeharding**
Thanks, @joshua-s for getting this started.

### CODE REVIEW

**Comment by joshowen** [comment]
Thanks @jakeharding for making this change.  Very much looking forward to its inclusion in a release.

**Comment by ferndot** [comment]
@mfogel could you review this please? It is a critical feature

**Comment by mdamien** [comment]
Hi, I'm one of the current temporary maintainer for the repo, I'm reviewing this as soon as possible, thanks for the PR. Would be nice to add other maintainers to this repo like you two.

File: .gitignore
Comment (mdamien) [inline .gitignore]: nitpick: I prefer this to not be in each repo but in your own global gitignore

**Comment by ferndot** [comment]
I'd be willing to assist!

**Comment by joshowen** [comment]
I'm also willing to help.

**Comment by mdamien** [comment]
Ok, @joshowen and @joshua-s, I'm gonna ask @mfogel to add you two as repo maintainer too. In the meantime, can you do the review of this PR and if I get your :+1:'s, I'm merging it :)

File: simple_email_confirmation/admin.py
Comment (ferndot) [inline simple_email_confirmation/admin.py]: This does not need to be a tuple.

```suggestion
admin.site.register(get_email_address_model(), EmailAddressAdmin)
```

File: simple_email_confirmation/models.py
Comment (ferndot) [inline simple_email_confirmation/models.py]: Could we move this to `__init__.py` like in `django.contrib.auth`?

File: simple_email_confirmation/models.py
Comment (ferndot) [inline simple_email_confirmation/models.py:131]: Maybe investigate setting `EmailAddressModel = get_email_address_model()` above this class so that the function is only called once. This pattern is used in: https://github.com/django/django/blob/master/django/contrib/auth/views.py#L28

**Review by ferndot — CHANGES_REQUESTED** [review]
Great work: it is looking really good! I have just a few suggestions.

Also, the coverage decreased a tiny bit. Could you bring it back up?

File: .gitignore
Comment (jakeharding) [reply .gitignore]: Sounds good. I will remove it.

File: simple_email_confirmation/admin.py
Comment (jakeharding) [reply simple_email_confirmation/admin.py]: Right, I never use a tuple in my admins, but it was a tuple before I made the change so I left it. I think it reads cleaner without the tuple so I will change it.

File: simple_email_confirmation/models.py
Comment (jakeharding) [reply simple_email_confirmation/models.py:131]: I was having trouble calling the method at the module level because the method would be called before the app loads all the models. I will chase it down a little further.

**Comment by jakeharding** [comment]
@joshua-s The coverage report showed all my lines covered so I don't know why the coverage went down. I will see what I can do to get it up.

File: simple_email_confirmation/models.py
Comment (jakeharding) [reply simple_email_confirmation/models.py]: 👍  Yep.

File: simple_email_confirmation/models.py
Comment (jakeharding) [reply simple_email_confirmation/models.py:131]: Django calls `get_user_model()` at import time with the use of a `require_ready` kwarg in the method call to `apps.get_model`. This kwarg was implemented in Django 1.11 with [this commit](https://github.com/django/django/commit/625cd5bcb315154ed28061fefaf639aff54da7c2#diff-e0a696d19bf764562a439d3080a30fda). Then used in `get_user_model` shortly after that in [this commit](https://github.com/django/django/commit/cb7bbf97a74fa7800865e3615f196ad65dc4f281#diff-82ec6fd74a1cab41408fbf5bf0998f58). Looks like both were part of the Django 1.11 release. 

In order to call `get_email_address_model` at the module level with 1.11 would take some work, but to call it with versions < 1.11 is not likely worth the effort. I definitely wouldn't include either in this PR. 😄

**Comment by jakeharding** [comment]
I found a couple easy tests to increase test coverage and adjusted how the base email address is stringified a bit according to the [Django docs](https://docs.djangoproject.com/en/1.9/_modules/django/utils/six/#python_2_unicode_compatible). Let me know what you think and thanks for all the great feedback! 🙇

File: simple_email_confirmation/models.py
Comment (ferndot) [reply simple_email_confirmation/models.py:131]: Sounds good, thanks for looking into it.

File: simple_email_confirmation/admin.py
Comment (ferndot) [inline simple_email_confirmation/admin.py]: We should update this since the function moved to `__init__.py`

```suggestion
from simple_email_confirmation import get_email_address_model
```

File: simple_email_confirmation/tests/tests.py
Comment (ferndot) [inline simple_email_confirmation/tests/tests.py]: Might be a better way to do this. Definitely import `get_email_address_model` directly from the package rather than `.models`.

```suggestion
from simple_email_confirmation import get_email_address_model
from simple_email_confirmation.models import EmailAddress, get_user_primary_email
from simple_email_confirmation.exceptions import (
    EmailConfirmationExpired, EmailIsPrimary, EmailNotConfirmed,
    EmailConfirmationExpired, EmailIsPrimary, EmailNotConfirmed,
)
from simple_email_confirmation.signals import (
    email_confirmed, unconfirmed_email_created, primary_email_changed,
    email_confirmed, unconfirmed_email_created, primary_email_changed,
)
```

**Review by ferndot — CHANGES_REQUESTED** [review]
Just 2 more tiny changes and this will have a green light from me!

File: simple_email_confirmation/tests/tests.py
Comment (jakeharding) [reply simple_email_confirmation/tests/tests.py]: Good catch on these! Sorry I missed them. I will update.

**Review by ferndot — APPROVED** [review]

**Comment by mdamien** [comment]
Thanks @joshua-s :), new release is coming

**Pull request merged**
